sys/include/net/netopt.h

```diff
@@ -269,6 +269,114 @@ typedef enum {
      */
     NETOPT_LAST_ED_LEVEL,
 
+    /**
+     * @brief   Get/Set preamble length as uint16_t in host byte order.
+     */
+    NETOPT_PREAMBLE_LENGTH,
+
+    /**
+     * @brief   Enable/disable CRC check
+     */
+    NETOPT_CRC,
+
+    /**
+     * @brief   Get random value
+     *
+     * Value is returned as uint32_t in host byte order.
+     */
+    NETOPT_RANDOM,
+
+    /**
+     * @brief   Enable/disable frequency hopping
+     */
+    NETOPT_FREQUENCY_HOP,
+
+    /**
+     * @brief   Get/Set frequency hopping period as uint8_t in host byte order.
+     */
+    NETOPT_FREQUENCY_HOP_PERIOD,
+
+    /**
+      * @brief   Enable/disable single packet reception.
+      * 
+      * If enabled, RX is turned off upon reception of a packet
+      */
+    NETOPT_SINGLE_RECEIVE,
+
+    /**
+     * @brief   Get/Set the reception timeout of a packet
+     * 
+     * Values are retrieved/passed as uint32_t in host byte order.
+     */
+    NETOPT_RX_TIMEOUT,
+
+    /**
+     * @brief   Get/Set the transmission timeout of a packet
+     * 
+     * Values are retrieved/passed as uint32_t in host byte order.
+     */
+    NETOPT_TX_TIMEOUT,
+
+    /** @brief   Get/Set the radio modem type.
+     *
+     * Values are passed as uint8_t and allowed values are:
+     * * 0: FSK modem
+     * * 1: LoRa modem (default)
+     *
+     * @see drivers_sx127x
+     */
+    NETOPT_MODEM_TYPE,
+
+    /** @brief   Get/Set the radio modulation bandwidth
+     * 
+     * Values are passed as uint8_t and allowed values are:
+     * * 0: 125kHz
+     * * 1: 250kHz
+     * * 2: 500kHz
+     *
+     * @see drivers_sx127x
+     */
+    NETOPT_BANDWIDTH,
+
+    /** @brief   Get/Set the radio spreading factor
+     * 
+     * Values are passed as uint8_t and allowed values are in the range 6..12.
```

Comment:
Please don't specify the range. You can check it in `_set()` and return `-EINVAL` for invalid values. This spreading factor range is specific to this radio.